### PR TITLE
clean up config.json reference in test-infra, update docs

### PR DESCRIPTION
**Pull request by krzyzacy**
/area this-is-unbearable
/assign @BenTheElder @spiffxp

### CODE REVIEW

File: config/README.md [hunk 1]
```diff
@@ -2,9 +2,19 @@
...
-
-Directory structure:
+## Directory structure:
 
 [jobs/](./jobs) : job configs for Kubernetes Prow deployment, potentially testgrid configs as well  
 [tests](./tests) : validation tests for the configs
+
+## Adding new jobs to Prow:
+
+1. Find or create a Org/Repo subdir under config/jobs.
```
Comment (spiffxp) [inline config/README.md]: Find or create an org/repo directory under `config/jobs`, eg: `config/jobs/kubernetes-sigs/kustomize` for jobs related to https://github.com/kubernetes-sigs/kustomize

Comment (krzyzacy) [reply config/README.md]: I like that I can just copy-paste your line :-)

**Comment by BenTheElder** [comment]
Thanks for this 👍

**Pull request merged**